pkg/downloader/chart_downloader.go

```diff
@@ -143,7 +143,39 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 
 func (c *ChartDownloader) getOciURI(ref, version string, u *url.URL) (*url.URL, error) {
 	var tag string
-	var err error
+
+	registryReference, err := registry.NewReference(u.Path)
```

Comment:
IMHO, we will need to document the below logic

Reply:
When you say document, do you mean comments or installation docs on https://helm.sh/ ?

Reply:
https://helm.sh/docs/topics/registries/ I think

The thought was to add to the examples e.g. installing by digest. And perhaps a "version specification" section which documents the `--version` and `:tag` and `@digest` options and precedence with respect to `helm pull`, `helm install`, etc.

(This PR can be merged without docs to clarify, just docs would make the contribution much more valuable IMHO)

Reply:
Sounds good. I'll put together a PR for the docs

Reply:
https://github.com/helm/helm-www/pull/1629

Reply:
Nice, thanks!